server/src/frameworks/Truffle/TruffleProject.ts

```diff
@@ -0,0 +1,267 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable @typescript-eslint/no-var-requires */
+/* eslint-disable no-empty */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { execSync } from "child_process";
+import { existsSync } from "fs";
+import _ from "lodash";
+import path from "path";
+import semver from "semver";
+import { DidChangeWatchedFilesParams } from "vscode-languageserver-protocol";
+import { OpenDocuments, ServerState } from "../../types";
+import { isRelativeImport } from "../../utils";
+import { directoryContains } from "../../utils/directoryContains";
+import { CompilationDetails } from "../base/CompilationDetails";
+import { BuildInputError } from "../base/Errors";
+import { FileBelongsResult, Project } from "../base/Project";
+import { Remapping } from "../base/Remapping";
+import { getDependenciesAndPragmas } from "../shared/crawlDependencies";
+
+enum Status {
+  NOT_INITIALIZED,
+  INITIALIZING,
+  INITIALIZED_SUCCESS,
+  INITIALIZED_FAILURE,
+}
+
+export class TruffleProject extends Project {
+  public priority = 3;
+  public sourcesPath!: string;
+  public testsPath!: string;
+  public globalNodeModulesPath!: string;
+  public remappings: Remapping[] = [];
+  public initializeError?: string;
+  public resolvedSolcVersion?: string;
+  public status: Status = Status.NOT_INITIALIZED;
+
+  constructor(
+    serverState: ServerState,
+    basePath: string,
+    public configPath: string
+  ) {
+    super(serverState, basePath);
+  }
+
+  public id(): string {
+    return this.configPath;
+  }
+
+  public frameworkName(): string {
+    return "Truffle";
+  }
+
+  public async initialize(): Promise<void> {
```

Comment:
Do we trigger a reinitialization on each config change?

Reply:
Yes, that logic is on the `onWatchedFilesChanges` method of this class.

Reply:
I pushed a change to get to the comparison check on `onWatchedFilesChanges` as the uri had `file://` as a prefix.

Is there a better way of doing that comparison with inbuilt vscode functions?

Reply:
You could use `URI.parse(<uri>).fsPath`. The import is `import { URI } from "vscode-uri";`